src/scheduler/pbs_sched.cpp

```diff
@@ -530,204 +506,117 @@ close_server(sched_svrconn *sconn)
-				   is_primary ? "primary" : "secondary", errmsg);
-			die(-1);
-		}
-		goto rerr;
+	/* free qrun_list */
+	for (i = 0; i < qrun_list_size; i++) {
+		if (qrun_list[i].jid != NULL)
+			free(qrun_list[i].jid);
 	}
+	free(qrun_list);
 
-	PBSD_FreeReply(reply);
-	return sock;
+	sched_svr_init();
 
-rerr:
-	pbs_disconnect(sock);
-	PBSD_FreeReply(reply);
-	return -1;
+	clust_primary_sock = -1;
+	clust_secondary_sock = -1;
 }
 
 /**
- * @brief connect to given server, create and add server struct in servers list
- *        also add secondary connection to poll list
+ * @brief connect to all the servers configured. Also add secondary connections
+ *              of individual servers secondary sd's to the poll list
  *
- * @param[in] svrhost - address of server in host[:port] format
  *
  * @return void
  */
 static void
-connect_server(const char *svrhost)
+connect_svrpool()
 {
-	int primary_sock = -1;
-	int secondary_sock = -1;
-	sched_svrconn *svr;
-	void *svrs_newarr;
-
-	if (svrhost == NULL)
-		return;
-
-	while (primary_sock < 0 || secondary_sock < 0) {
-		if (primary_sock < 0) {
-			primary_sock = connect_server_helper(svrhost, 1);
-			if (primary_sock < 0) {
+	int i;
+	svr_conn_t *svr_conns;
+	int registered = 0;
+
+
+	while (clust_primary_sock < 0 || clust_secondary_sock < 0 || registered == 0) {
+		if (clust_primary_sock < 0) {
+			clust_primary_sock = pbs_connect(NULL);
+			if (clust_primary_sock < 0) {
 				/* wait for 2s for not to burn too much CPU, and then retry connection */
 				sleep(2);
 				continue;
 			}
 		}
-		secondary_sock = connect_server_helper(svrhost, 0);
-		if (secondary_sock < 0) {
+		clust_secondary_sock = pbs_connect(NULL);
+		if (clust_secondary_sock < 0) {
 			/* wait for 2s for not to burn too much CPU, and then retry connection */
 			sleep(2);
 			continue;
 		}
-	}
-	if (tpp_em_add_fd(poll_context, secondary_sock, EM_IN | EM_HUP | EM_ERR) < 0) {
-		log_errf(errno, __func__, "Couldn't add secondary connection to poll list for server %s", svrhost);
-		pbs_disconnect(primary_sock);
-		pbs_disconnect(secondary_sock);
-		die(-1);
-	}
 
-	svr = static_cast<sched_svrconn *>(malloc(sizeof(sched_svrconn)));
-	if (!svr) {
-		log_err(errno, __func__, MEM_ERR_MSG);
-		die(-1);
+		if (pbs_register_sched(sc_name) == 0) {
+			log_errf(pbs_errno, __func__, "Couldn't register the Scheuduler %s with the configured servers", sc_name);
+			/* wait for 2s for not to burn too much CPU, and then retry connection */
+			sleep(2);			
+			continue;
+		} else
+			registered = 1;
+		
 	}
 
-	svr->svrhost = strdup(svrhost);
-	if (!svr->svrhost) {
-		free(svr);
-		log_err(errno, __func__, MEM_ERR_MSG);
-		die(-1);
-	}
-	svr->primary_sock = primary_sock;
-	svr->secondary_sock = secondary_sock;
-	svrs_newarr = add_ptr_to_array(servers, svr);
-	if (svrs_newarr == NULL) {
-		close_server(svr);
-		die(-1);
+	svr_conns =  static_cast<svr_conn_t *>(get_conn_servers(clust_secondary_sock));
```

Comment:
Same comment, shouldn't we be passing -1?

Reply:
Same as my earlier comment. This is the correct usage.